codelabs/codelab-02/codelab2README.md

```diff
@@ -0,0 +1,501 @@
+
+The tests can be run like so:
+
+```
+$ python test.py
+```
+
+Note that if you kill the testing program, it won't clean up after itself. If you end up with extra testing buckets, you can quickly remove them with the following command:
+
+```
+$ aws s3 ls | grep "test-bucket" | cut -f3 -d ' ' | xargs -I {} aws s3 rb 's3://{}'
+```
+
+Feel free to install any packages you would like to use. Just make sure to add them to the `requirements.txt` file (run `pip freeze > requirements.txt`).
+
+#### Host a Static Website
+
+As mentioned in class, you can use S3 to host static content, like static websites. Let's do that for the STICs website!
+
+Now that you've finished the assignment, you have a tool that you could use to upload static content for a website, however the tool does not set the Content-Type of each file when they are uploaded (they default to a byte stream type). If you tried to open an HTML file in your browser that was hosted on S3 with this Content-Type, then your browser would just download the file instead of serving it (it doesn't use the file ending to determine the type of file, as you might expect, it instead uses the Content-Type header in the response).
+
+Therefore you have two options: 1) update your tool to detect and set the Content-Type header (use a library like [`python-magic`](https://github.com/ahupp/python-magic)). Otherwise, you can use the `aws s3 sync` command.
+
+You will want to clone the source code from it's GitHub repository here: https://github.com/UMD-CS-STICs/UMD-CS-STICs.github.io
+
+```
+$ git clone https://github.com/UMD-CS-STICs/UMD-CS-STICs.github.io
+```
+
+Then, create a new bucket specifically for this site, such as `cmsc389l-<directory id>-website`. Then, enable website hosting from that bucket:
+
+```
+$ aws s3 website s3://cmsc389l-colink-website —index-document index.html
+```
+
+##### Option 1: upload.py
+
+Assuming that your tool will properly set the Content-Type header, then you can copy the source code into the root directory of this new bucket using your tool:
+
+```
+$ python upload.py UMD-CS-STICs.github.io —bucket cmsc389l-colink-website —acl public-read
+```
+
+##### Option 2: s3 sync
+
+If not, you can use the `s3 sync` tool instead. Make sure to sync the contents of the STICs site into the root of the bucket.
+
+##### View the site!
+
+To view the site, browse to the S3 URL of `index.html` and you will see the full STICs site!
+
+```
+https://s3.amazonaws.com/cmsc389l-colink-website/index.html
+```
+
+Submit a screenshot of the front page of the STICs site hosted on your S3. Make sure to include the URL in the screenshot (don't just submit a screenshot of [sticsumd.com](http://sticsumd.com (http://sticsumd.com/))!).
+
+### Submission
+
+You will submit a zipped directory containing your script, `upload.py`, plus an updated `requirements.txt` if you installed any packages. You will also include the screenshot of the STICs homepage and your audit log.
```

Comment:
they will have installed python-magic/etc., so they will need to provide the Pipfile